Backend/AlgorithmsManager.py

```diff
@@ -389,6 +389,67 @@ def parse_city_data(self, entry):
         city_data["race_multiple"] = entry[29]
 
         return city_data
+        
+    def get_user_by_username(self, username):
+        print("\nValidating user")
+        try:
+        # def __init__(self, id, ids, uname, signup, loc, protected, folcount, fricount):
+        # self.user_PK = id
+        # self.user_id = ids
+        # self.username = uname
+        # self.signup_date = signup
+        # self.location_id = loc
+        # self.protected = protected
+        # self.followers_count = folcount
+        # self.friends_count = fricount
+            ur = self.twitter.get_user(username)
+            user = DataStructures.User(None, ur.id, ur.screen_name, ur.name, ur.created_at, ur.location,
+                                        ur.protected, ur.followers_count, ur.friends_count)
+            res = []
+            res.append(user.__dict__)
+            print("\nValidation success")
+            return res
+        
+        except:
+            print("Validation fails")
+            return []
+
+    # this function returns a list of retweeters
+    #   args:   username
+    #           num_tweets (optional): how many recent tweets to search through
+    #           num_retweets (optional): how many retweeters per tweet
+    #           sorted (optional): value of 1 sorts the list by number of retweets 
+    #   return: a list of dictionary
+    def get_most_frequent_retweeters(self, username, num_tweets=20, num_retweets=20, sorted=1):
+
+        retweetData = self.twitter.most_recent_retweeters(username, num_tweets=num_tweets, num_retweets=num_retweets)
+        
+        # get the list of all retweeters
+        retweeter_history = []
+        for tweet in retweetData:
+            innerList = tweet['retweeters']
+            for username in innerList:
+                retweeter_history.append(username)
```

Comment:
I'm pretty sure this retweeter_history list shouldn't include the same username twice. However, if a certain other user retweeted multiple recent tweets by the given user, wouldn't that user's username be part of the 'retweeters' lists of multiple 'tweet' dictionaries in 'retweetData'? Then, wouldn't that user's username end up being appended to retweeter_history multiple times?

Reply:
The number of times a username appears in retweeter_history list is how often that username retweets the given user, _across multiple recent tweets_ (only one retweet per tweet). Does this resolve your question?